app/Models/Multiplayer/Room.php

```diff
@@ -103,18 +107,48 @@ public static function preloadRecentParticipants(Collection $rooms)
         }
     }
 
-    public static function search(array $rawParams)
+    public static function responseJson(array $rawParams): array
+    {
+        $typeGroup = $rawParams['type_group'] ?? null;
+
+        $search = static::search($rawParams, 50);
+
+        [$rooms, $hasMore] = $search['query']->with([
+            'playlist.beatmap',
+            'host',
+        ])->getWithHasMore();
+
+        $rooms->each->findAndSetCurrentPlaylistItem();
+        $rooms->loadMissing('currentPlaylistItem.beatmap.beatmapset');
+
+        $response['rooms'] = json_collection($rooms, new RoomTransformer(), ['current_playlist_item.beatmap.beatmapset', 'difficulty_range', 'host', 'playlist_item_stats']);
+
+        if ($typeGroup !== null) {
+            $response['type_group'] = $typeGroup;
+        }
+
+        $nextCursor = $hasMore ? $search['cursorHelper']->next($rooms) : null;
+
+        return array_merge($response, cursor_for_response($nextCursor));
+    }
+
+    public static function search(array $rawParams, ?int $maxLimit = null)
     {
         $params = get_params($rawParams, null, [
             'category',
             'limit:int',
+            'max_limit:int',
             'mode',
+            'season_id:int',
             'sort',
             'type_group',
             'user:any',
         ], ['null_missing' => true]);
 
+        $maxLimit ??= 250;
+        $maxLimit = clamp($maxLimit, 1, 250);
```

Comment:
not needed (it should only be set internally)

Reply:
won't the issue mentioned in https://github.com/ppy/osu-web/pull/9725#discussion_r1095453624 stay then?

Reply:
only if it comes from external variable (rooms index endpoint passes all params by default)

Reply:
oh right, didn't notice that